CMakeLists.txt

```diff
@@ -3,14 +3,31 @@ project(exit2destination LANGUAGES C CXX)
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -pedantic -std=c++11")
 
-find_package(Boost REQUIRED)
+#find_package(Boost REQUIRED)
 find_package(Threads REQUIRED)
 
 # Vendored Libosmium
-list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/third_party/libosmium/cmake")
-set(OSMIUM_INCLUDE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/third_party/libosmium/include")
-find_package(Osmium REQUIRED COMPONENTS pbf)
+if (ENABLE_MASON)
+    set(MASON_COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/.mason/mason)
+    include(${CMAKE_CURRENT_SOURCE_DIR}/.mason/mason.cmake)
+
+    mason_use(libosmium VERSION 2.10.3 HEADER_ONLY)
+    include_directories(${MASON_PACKAGE_libosmium_INCLUDE_DIRS})
+    set(OSMIUM_INCLUDE_DIRS ${MASON_PACKAGE_libosmium_INCLUDE_DIRS})
+
+    mason_use(protozero VERSION 1.4.2 HEADER_ONLY)
+    include_directories(${MASON_PACKAGE_protozero_INCLUDE_DIRS})
+    set(PROTOZERO_INCLUDE_DIRS ${MASON_PACKAGE_protozero_INCLUDE_DIRS})
+else()
+    # expects a vendored libosmium at /third_party
+    list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/third_party/libosmium/cmake")
+    set(OSMIUM_INCLUDE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/third_party/libosmium/include")
+    find_package(Osmium)
+endif()
+
+find_package(ZLIB REQUIRED)
+include_directories(${ZLIB_INCLUDE_DIRS})
```

Comment:
These two lines have to go.

We don't depend on zlib. Libosmium does.

The `find_package(Osmium)` searches for zlib and adds the include dirs and libraries.
I don't know what mason does but it should do the same.